Core/src/org/sleuthkit/autopsy/commonfilesearch/CommonAttributeSearchResults.java

```diff
@@ -48,32 +58,103 @@
      * This is a convenience method - you can also iterate over
      * <code>getValues()</code>.
      *
-     * @param isntanceCound key
+     * @param instanceCount key
      * @return list of values which represent matches
      */
     List<CommonAttributeValue> getAttributeValuesForInstanceCount(Integer instanceCount) {
         return this.instanceCountToAttributeValues.get(instanceCount);
     }
 
- /**
-     * Get an unmodifiable collection of values, indexed by number of 
-     * grandchildren, which represents the common attributes found in the 
+    /**
+     * Get an unmodifiable collection of values, indexed by number of
+     * grandchildren, which represents the common attributes found in the
      * search.
+     *
      * @return map of sizes of children to list of matches
-     */    
-public Map<Integer, List<CommonAttributeValue>> getMetadata() {
+     */
+    public Map<Integer, List<CommonAttributeValue>> getMetadata() throws EamDbException {
+        if(this.percentageThreshold == 0){
+            return Collections.unmodifiableMap(this.instanceCountToAttributeValues);
+        } else {
+            return this.getMetadata(this.percentageThreshold);
+        }
+    }
+    
+    /**
+     * Get an unmodifiable collection of values, indexed by number of
+     * grandchildren, which represents the common attributes found in the
+     * search.
+     * 
+     * Remove results which are not found in the portion of available data 
+     * sources described by minimumPercentageThreshold.
+     * 
+     * @return metadata
+     */
+    private Map<Integer, List<CommonAttributeValue>> getMetadata(int minimumPercentageThreshold) throws EamDbException {
+        
+        if(minimumPercentageThreshold == 0){
+            return Collections.unmodifiableMap(this.instanceCountToAttributeValues);
+        }
+        
+        CorrelationAttributeInstance.Type fileAttributeType = CorrelationAttributeInstance
+                .getDefaultCorrelationTypes()
+                .stream()
+                .filter(filterType -> filterType.getId() == CorrelationAttributeInstance.FILES_TYPE_ID)
+                .findFirst().get();
+        
+        EamDb eamDb = EamDb.getInstance();
+        
+        Map<Integer, List<CommonAttributeValue>> itemsToRemove = new HashMap<>();
+        
+        for(Entry<Integer, List<CommonAttributeValue>> listOfValues : Collections.unmodifiableMap(this.instanceCountToAttributeValues).entrySet()){
+            
+            final Integer key = listOfValues.getKey();
+            final List<CommonAttributeValue> values = listOfValues.getValue();
+            
+            for(CommonAttributeValue value : values){
+                
+                int frequencyPercentage = eamDb.getFrequencyPercentage(new CorrelationAttributeInstance(fileAttributeType, value.getValue()));
+                
+                if(frequencyPercentage < minimumPercentageThreshold){
```

Comment:
Is this comparison backwards? I thought the idea was to exclude files that have a frequency greater than the threshold - i.e., exclude files that are common and appear many times in the central repo.